lib/arel/visitors/spanner.rb

```diff
@@ -22,6 +33,29 @@ def bind_block
       end
 
       # rubocop:disable Naming/MethodName
+      def visit_Arel_Nodes_OptimizerHints o, collector
+        o.expr.each do |v|
```

Comment:
when will we get multiple `v` from `o`? so `v` is a single optimizer hint?

Reply:
Correct, `v` is a single optimizer hint. So you would get multiple `v` if the user would call the following:

```ruby
Singer.optimizer_hints("max_staleness: 5.45").optimizer_hints("exact_staleness: 10").find_by(id: 1)
```

In this case, the last optimizer staleness hint would take precedence, as these are mutually exclusive. But the plan is also to support other hints that are not mutually exclusive with the staleness hints, so we need to iterate over the values that we receive.

One possibility would however be to raise an error if we receive multiple staleness hints for one query, instead of the current behavior to let the last one take precedence.

Reply:
Thanks for the explanation. I am okay with the current approach.